kotlin/codegen/build.gradle.kts

```diff
@@ -109,3 +109,5 @@ afterEvaluate {
     }
   }
 }
+
+tasks.named("assemble") { dependsOn(tasks.named("shadowJar")) }
```

Comment:
Why is this needed? shadowJar is already configured above this for publications

```
> Task :kotlin:codegen:jar
> Task :kotlin:codegen:shadowJar
> Task :kotlin:tests:kaptGenerateStubsTestKotlin
> Task :kotlin:tests:kaptTestKotlin
```

shadowed kotlinpoet/kotlinx-metadata classes are present too 

![image](https://user-images.githubusercontent.com/1361086/92255423-4c452780-eea0-11ea-90e4-f7ab891704be.png)

The problem here appears to be that kotlinpoet itself is getting shaded

![image](https://user-images.githubusercontent.com/1361086/92255657-a34afc80-eea0-11ea-859b-3c5b450b7fea.png)

I tried with your changes, and the same issue is still present

![image](https://user-images.githubusercontent.com/1361086/92255871-ec02b580-eea0-11ea-8630-afe2aaeb116d.png)

The proper fix here is to figure out why this block is excluding kotlinpoet's core artifact from the pom and shading it. This PR just updates the shadow plugin dependency but doesn't fix anything. We probably missed this in the switch to gradle because the kotlin gradle plugin often just uses compiled class files from their local project's build dir directly rather than packaged jars (sometimes annoying quirk of how their incremental compilation works).

![image](https://user-images.githubusercontent.com/1361086/92256017-1f454480-eea1-11ea-9aff-987834ce67ee.png)

One quick fix is to add `exclude(group = "com.squareup", module = "kotlinpoet")` to the shaded kotlinpoet deps to ensure it doesn't shade the core artifact, but still leaves the `runtime` issue. This appears to be a long-standing issue in the shadow plugin: https://github.com/johnrengelman/shadow/issues/321. We can fix this with a `withXml` adjustment. I've opened #1214 to do these

Reply:
This is needed because it ran too late and the source code was not updated.
So the code still referenced the non shadowed one and failed.